readthedocs/builds/version_slug.py

```diff
@@ -64,11 +64,24 @@ def slugify(self, content):
         if not content:
             return ''
         slugified = ''
+        content = content.lower()
         for char in content:
             if char not in self.allowed_chars:
                 slugified += self.placeholder
             else:
                 slugified += char
+
+        # Do not start and end in punctuation.
+        slug_length = len(slugified)
+        diff = 1
+        while diff > 0:
+            for char in self.allowed_punctuation:
+                slugified = slugified.strip(char)
+            diff = slug_length - len(slugified)
+            slug_length = len(slugified)
+
+        if not slugified:
+            return self.fallback_slug
```

Comment:
Good idea, that is addressed with the latest push. I've not used regex.sub before that's why I didn't think about it. It reduced the code quite a bit. Thanks for the suggestion!